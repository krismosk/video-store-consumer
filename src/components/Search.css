.search-bar {
  margin-top: 1em;
  background: #fff;
  padding: 1em;
  width: 95%;
  border-radius: 10em;
  transition: box-shadow 300ms ease;
  border-radius: 6px;
  border: 1px solid rgba(137, 149, 199, .2);
  box-shadow: inset 0 2px 0 0 rgba(0, 0, 0, .075) !important;
  margin-left: 10px;
  margin-right: 10px;
  margin-bottom: 20px;
}

.search-bar--label {
  font-size: 1.5em;
  font-weight: bolder;
  margin-left: 10px;
}

.elevation-3 {
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12) !important;
}
.notification {
  padding: 10px;
  color: white;
}
.error {
  background: red;
}

.success{
  background: green;
}

.warning {
  background: yellow;
}

