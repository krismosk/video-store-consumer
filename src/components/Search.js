import React, { Component } from 'react';
import PropTypes from 'prop-types';
// npm install bootstrap
// import 'bootstrap/dist/css/bootstrap.min.css';
import './Search.css';
import axios from 'axios';
import MovieDb from './MovieDb'
import { nullLiteral } from '@babel/types';

class SearchBar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      searchTerm: '',
      movieToAdd: undefined,
      results: [],
    };
  }
// const SearchBar = ({ searchTerm, searchChangeCallback }) => {

searchDatabase = (searchTerm) => {
  
  console.log(`https://api.themoviedb.org/3/search/movie?api_key=a4e96005d8a3d373dc3607963dc38e8f&query=${searchTerm}`);
  axios.get(`https://api.themoviedb.org/3/search/movie?api_key=a4e96005d8a3d373dc3607963dc38e8f&query=${searchTerm}`)
  .then((response) => {
    this.setState({
      searchTerm: searchTerm,
      results: response.data.results,
    });
    this.displayResults();
  })
  .catch((error) => {
    console.log('error');
  });
}

displayResults = () => {
  console.log(this.state.searchTerm);
  console.log(this.state.results);
  console.log(this.state.results[0]);
  const resultList = (this.state.results).map((movie) => {
    return <MovieDb title={movie.title} overview={movie.overview} release_date={movie.release_date} image_url={movie.poster_path} external_id={movie.id}/>
  });
  return resultList;
}

render () {

return (
  <section>
      <div>
        <label className="search-bar--label" htmlFor="searchBar">Search Movie Database: </label>
      </div>
      <input
        onChange={(event) => { this.searchDatabase(event.target.value) }}
        value={this.searchTerm}
        name="searchBar"
        id="searchBar"
        className="search-bar"
      />
      <div>{this.displayResults()}</div>
    </section>
  )
}
};

SearchBar.propTypes = {
  // searchChangeCallback: PropTypes.func.isRequired,
  // searchTerm: PropTypes.string.isRequired,
};

export default SearchBar;