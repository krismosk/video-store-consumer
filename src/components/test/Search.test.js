import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from '../Search';

const search = <Search />

describe("Search", () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render ( search );

    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
})