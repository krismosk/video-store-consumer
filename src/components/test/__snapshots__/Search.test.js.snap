// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <section>
    <div>
      <label
        class="search-bar--label"
        for="searchBar"
      >
        Search Movie Database: 
      </label>
    </div>
    <input
      class="search-bar"
      id="searchBar"
      name="searchBar"
      value=""
    />
    <div />
  </section>
</DocumentFragment>
`;
